alerting/src/main/kotlin/org/opensearch/alerting/chainedAlertCondition/resolvers/ChainedAlertRPNResolver.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.chainedAlertCondition.resolvers
+
+import org.opensearch.alerting.chainedAlertCondition.tokens.CAExpressionOperator
+import org.opensearch.alerting.chainedAlertCondition.tokens.CAExpressionToken
+import org.opensearch.alerting.chainedAlertCondition.tokens.ChainedAlertExpressionConstant
+import org.opensearch.alerting.chainedAlertCondition.tokens.ExpressionToken
+import java.util.Stack
+
+/**
+ * Solves the Trigger Expression using the Reverse Polish Notation (RPN) based solver
+ * @param polishNotation an array of expression tokens organized in the RPN order
+ */
+class ChainedAlertRPNResolver(
```

Comment:
Wondering why we did not extend on the current ``TriggerExpressionResolver``

Reply:
i had started out to do that but midway realized I had a much simpler use case which relied on RPN resolution and needed a different method too.

We can explore extracting commonn logic out but method params were little different and I can do a follow up. Test classes have been added to verify the chained alerts parsing

Reply:
Thanks can we add to our backlog in the meta

Reply:
https://github.com/opensearch-project/alerting/issues/1005